google/cloud/storage/bucket.py

```diff
@@ -726,6 +734,53 @@ def user_project(self):
         """
         return self._user_project
 
+    @property
+    def generation(self):
+        """Retrieve the generation for the bucket.
+
+        :rtype: int or ``NoneType``
+        :returns: The generation of the bucket or ``None`` if the bucket's
+                  resource has not been loaded from the server. Only buckets
+                  with soft delete policies have generations; buckets without
+                  such a policy will also have ``None`` in this field. See
+                  the Bucket.soft_delete_policy property.
```

Comment:
Could we check if this is still true? I see go/gcs-soft-delete-bucket-restores-user-guide states that generation exists for every bucket, regardless of whether soft delete is enabled.

Reply:
Good catch. Not sure how I got my wires so crossed here. It does appear to exist regardless of whether soft delete is enabled. I'll fix it.